### DISCUSSION TITLE
[clickhouse] support customizable clickhouse database name

### DISCUSSION THREAD

**Discussion by void-main** [langfuse/langfuse#4843] [Ideas]
### Describe the feature or potential improvement

According to the set-hosting doc on [clickhose](https://langfuse.com/self-hosting/infrastructure/clickhouse):

> Please note, that Langfuse uses the default schema and default cluster. Get in touch with us on [Discord](https://langfuse.com/discord) or contact the maintainers at [support@langfuse.com](mailto:support@langfuse.com) if this limitation is blocking.

We're using a clickhouse service form a cloud service provider, and the `default` database is reserved, so we could not (and should not) use `default` database. This is blocking our deployment.

Any chances we could use another database instead of `default`?

**Comment by Steffen911** [comment]
@void-main Could you share the cloudprovider that you're using? For us this is a new restriction that we haven't encountered before so we'd like to understand whether this is a common restriction.

There may be a workaround that would involve some effort on your side. If you adjust https://github.com/langfuse/langfuse/tree/main/packages/shared/clickhouse/migrations locally to point to a different cluster and database it may work for your case. These adjustments would adjust the migrations we run and in addition you can configure https://github.com/langfuse/langfuse/blob/main/packages/shared/src/server/clickhouse/client.ts#L13 to overwrite the ClickHouse client that we use.

If there is demand from the community for this, we'll also consider standardizing this and make it configurable across the application.

**Comment by bharath-muthineni** [comment]
Hey @Steffen911 , 
We have a clickhouse hosted at our org. level and the cluster name has been fixed as `cluster`. We need similar support to customize the name of the cluster from `default` to something something different. What do you suggest?

**Comment by bharath-muthineni** [comment]
Hello @Steffen911 , were you able to check on this?

**Comment by peske** [comment]
Why actually Docker compose files include `CLICKHOUSE_DB` environment variable (https://github.com/langfuse/langfuse/issues/4888)? It took me hours to try to deploy like that. Needless to say, with no luck.

Also, frankly speaking, discussion if Langfuse should support customizable DB name is bit strange. Of course it should! Besides the obvious, sanity reason, here's one concrete (mine at the moment): I have quite a powerful ClickHouse cluster, used by several other services, but still more than capable for Langfuse. And now I cannot use it because the DB name? Not sure what to say...

**Comment by peske** [comment]
I've created PR: https://github.com/langfuse/langfuse/pull/4890

**Comment by Steffen911** [comment]
Will be resolved via https://github.com/langfuse/langfuse/issues/4888

**Comment by gavinshark** [comment]
Could you please add the feature in helm? The database name can not be specified in helm by default.

**Reply by Steffen911** [reply]
@gavinshark Do you use the ClickHouse deployment that is included in the helm chart or an external deployment? If it's the former I'd like to understand why you have this requirement. If you're using an external ClickHouse, you could work around this restriction by setting CLICKHOUSE_DB via the `additionalEnv` parameters.
